File: Daily Challenge/C/Row & Column Intersection 20.8.2025.c
#include <stdio.h>
int main() {
    int r;
    scanf("%d", &r);
    int mat[r][r];
    for(int i=0;i<r;i++)
        for(int j=0;j<r;j++)
            scanf("%d", &mat[i][j]);

    for(int i=0;i<r;i++) {
        int row = 1, col = 1;
        for(int k=0;k<r;k++) {
            if(k!=i && mat[i][k]!=0) row=0;
            if(k!=i && mat[k][i]!=1) col=0;
        }
        if(row && col) {
            printf("%d", i+1);
            return 0;
        }
    }
    printf("-1");
    return 0;
}

File: Daily Challenge/Java/Row & Column Intersection 20.8.2025.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int r = sc.nextInt();
        int[][] mat = new int[r][r];
        for(int i=0;i<r;i++)
            for(int j=0;j<r;j++)
                mat[i][j] = sc.nextInt();

        for(int i=0;i<r;i++) {
            boolean row = true, col = true;
            for(int k=0;k<r;k++) {
                if(k!=i && mat[i][k]!=0) row=false;
                if(k!=i && mat[k][i]!=1) col=false;
            }
            if(row && col) {
                System.out.println(i+1);
                return;
            }
        }
        System.out.println(-1);
    }
}